tests/rptest/tests/consumer_group_test.py

```diff
@@ -1505,3 +1506,192 @@ def verify_all_consumers_are_present():
+        self.id = id
+        self.group = group
+        self.topic = topic
+        self.consumer_thread = threading.Thread(
+            name=f'consumer-{id}',
+            target=lambda this: this.loop(),
+            args=[self])
+        self.stopped = threading.Event()
+
+        self.logger = logger
+        self.consumer_thread.daemon = True
+        self.last_committed = -1
+        self.lock = threading.Lock()
+        self.restarted = threading.Event()
+        self.partition_id = partition_id
+        self.next = 0
+        self.consumer_thread.start()
+
+    def stop(self):
+        self.logger.info(f"stopping consumer with id: {self.id}")
+        self.stopped.set()
+        self.consumer_thread.join()
+
+    def is_stopped(self):
+        return self.stopped.is_set()
+
+    def create_consumer_client(self):
+        self.consumer = Consumer(
+            {
+                "group.id": self.group,
+                "client.id": f"consumer-{self.id}",
+                'bootstrap.servers': self.bootstrap_servers,
+                'auto.offset.reset': 'earliest',
+                'enable.auto.offset.store': True,
+                'enable.auto.commit': False,
+            },
+            logger=self.logger)
+
+        self.consumer.assign([TopicPartition(self.topic, self.partition_id)])
+
+    def loop(self):
+        self.create_consumer_client()
+        self.logger.info(f"starting consumer with id: {self.id}")
+        while not self.stopped.is_set():
+            try:
+                to_commit = self.next
+                self.next += 1
+                ret = self.consumer.commit(offsets=[
+                    TopicPartition(self.topic,
+                                   self.partition_id,
+                                   offset=to_commit)
+                ],
+                                           asynchronous=False)
+                with self.lock:
+                    self.last_committed = ret[0].offset
+            except Exception as e:
+                self.logger.error(f"consumer {self.id} error - {e}")
+        self.logger.info(f"closing consumer with id: {self.id}")
+        self.consumer.close()
+
+    def get_last_committed(self):
+        with self.lock:
+            return self.last_committed
+
+
+class ConsumerGroupOffsetResetTest(RedpandaTest):
+    """
+    This tests simulates a large number of consumers trying to commit consumer 
+    group offsets. The test doesn't produce or fetch any messages, 
+    it just stress tests OffsetCommit requests and validates the final 
+    state of the consumer group.
+    """
+    def __init__(self, test_context):
+        super(ConsumerGroupOffsetResetTest,
+              self).__init__(test_context=test_context,
+                             num_brokers=3,
+                             extra_rp_conf={
+                                 "group_topic_partitions": 1,
+                                 "compacted_log_segment_size": 1024 * 1024,
```

Comment:
did you mean compacted_log_segment_size or log_segment_size, double checking since all the data lands in the same segment with 128_MiB seg size I think.

Reply:
i am only interested in the `compacted_log_segment_size` as this is what applies to the `__consumer_offsets` topic.

Reply:
oo TIL..

```
    if (config().has_overrides() && config().get_overrides().segment_size) {
        result = *config().get_overrides().segment_size;
    } else {
        // no overrides use defaults
        result = config().is_compacted()
                   ? _manager.config().compacted_segment_size()
                   : _manager.config().max_segment_size();
    }

```

`compacted_segment_size` config doc-string doesn't say that anywhere, so I thought segment_size is a must.